src/lro/azureAsyncPolling.ts

```diff
@@ -21,15 +25,12 @@ function isAzureAsyncPollingDone(rawResponse: FullOperationResponse) {
 }
 
 export function processAzureAsyncOperationResult<TResult>(
-  restrieveResource: (path?: string) => Promise<LROResult<TResult>>,
+  lro: LRO<TResult>,
```

Comment:
passing the whole object is needed so `retrieveAzureAsyncResource` can access `this` internaly.